application/src/main/java/com/sanctionco/thunder/resources/UserResource.java

```diff
@@ -165,10 +165,28 @@ public Response updateUser(@Auth Key key,
           .entity("Unable to validate user with provided credentials.").build();
     }
 
+    // Determine what verification information to use for the updated user object.
+    // If it's a new email address, reset verification status.
+    // If it's the same, keep the existing verification status.
+    boolean verified = email.equals(user.getEmail().getAddress())
+        && foundUser.getEmail().isVerified();
+
+    String verificationToken = email.equals(user.getEmail().getAddress())
+        ? foundUser.getEmail().getVerificationToken()
+        : null;
+
+    LOG.info("Using verified status: {} and token: {} for the updated user.",
```

Comment:
How will this react to a null value in the case that the found user is not verified?

Reply:
Good question - it should be fine with `null`, the log line will just print out `"null"` where it is supposed to be. There shouldn't be any NPEs here